module/VuFind/src/VuFind/Security/CspHeaderGenerator.php

```diff
@@ -198,4 +201,34 @@ public function getReportToHeader()
         $reportToHeader->setFieldValue(implode(', ', $groupsText));
         return $reportToHeader;
     }
+
+    /**
+     * Create NEL (Network Error Logging) header based on given configuration
+     *
+     * @return ?GenericHeader
+     */
+    public function getNetworkErrorLoggingHeader()
+    {
+        $nelHeader = new \Laminas\Http\Header\GenericHeader();
+        $nelHeader->setFieldName('NEL');
+        $nelData = [];
+
+        $nelConfig = $this->config->NetworkErrorLogging;
+        if ($reportTo = $nelConfig['report_to'] ?? null) {
+            $nelData['report_to'] = $reportTo;
+        } else {
+            return null;
+        }
+        $nelData['max_age'] = $nelConfig['max_age'] ?? 86400; // one day
+        if ($includeSubdomains = (bool)$nelConfig['include_subdomains'] ?? null) {
```

Comment:
I think you need to add parentheses around the coalescing clause or else operator precedence may not work as expected. (Same for the other one below).

Reply:
Actually I realized I had to get rid of the null coalescing operators since it has to handle falsy values and include them in the header.  failure_fraction = 0 is a valid param and the opposite of omitting it.  Doesn't matter for include_subdomains (default is false) but I think it's better to be consistent.